File: solutions/go/raindrops/2/raindrops.go
package raindrops

import (
    "strconv"
    "strings"
)

var soundMap = [3]struct{
    divisor	int
    sound	string
} {
    {3, "Pling"},
    {5, "Plang"},
    {7, "Plong"},
}

func Convert(number int) string {
	var builder strings.Builder
    for _, mapping := range soundMap {
        if number % mapping.divisor == 0 {
            builder.WriteString(mapping.sound)
        }
    }
    if builder.Len() == 0 {
        builder.WriteString(strconv.Itoa(number))
    }
    return builder.String()
}

File: solutions/c/raindrops/2/raindrops.c
#include "raindrops.h"

#include <string.h>
#include <stdio.h>

void convert(char *const result, const int drops) {
    if (drops % 3 == 0) {
        strcat(result, "Pling");
    }
    if (drops % 5 == 0) {
        strcat(result, "Plang");
    }
    if (drops % 7 == 0) {
        strcat(result, "Plong");
    }
    if (!*result) {
        sprintf(result, "%d", drops);
    }
}